src/js/line.js

```diff
@@ -0,0 +1,517 @@
+        },
+        svgOptions: {
+            width: 700,
+            height: 500
+        },
+        lineOptions: {
+            // An array of colors to use for lines drawn from the data
+            // Or, a d3 color scale that's generated based off an array of colors
+            colors: null,
+            // Padding of the chart elements inside the SVG container
+            padding: 50,
+            // interpolation mode for chart line
+            // see line.interpolate at https://github.com/mbostock/d3/wiki/SVG-Shapes
+            // generally, "linear" for sharp lines, "cardinal" for smooth
+            interpolation: "linear",
+            // Whether or not to add an area fill under the chart line
+            addArea: false,
+            // Whether or not to add a point to each datapoint forming the line
+            addPoints: false,
+            pointRadius: 2,
+            // In milliseconds
+            transitionLength: 2000
+        },
+        styles: {
+            svg: "floe-ca-lineChart-chart",
+            chartLinePoint: "floe-ca-lineChart-chartLinePoint"
+        },
+        selectors: {
+            title: ".floec-ca-lineChart-title",
+            description: ".floec-ca-lineChart-description",
+            svg: ".floec-ca-lineChart-line",
+            yAxis: ".floec-ca-lineChart-y-axis",
+            xAxis: ".floec-ca-lineChart-x-axis",
+            chartLine: ".floec-ca-lineChart-chartLine",
+            chartLinePointGroup: ".floec-ca-lineChart-chartLinePointGroup",
+            chartLinePoint: ".floec-ca-lineChart-chartLinePoint",
+            chartLineArea: ".floec-ca-lineChart-chartLineArea"
+        },
+        events: {
+            onChartCreated: null,  // Fire when the line is created. Ready to register D3 DOM event listeners,
+            onChartRedrawn: null // Fire when the chart is redrawn.
+        },
+        listeners: {
+            "onCreate.create": {
+                funcName: "floe.chartAuthoring.lineChart.chart.create",
+                args: ["{that}"]
+            }
+        },
+        modelListeners: {
+            dataSet: {
+                funcName: "{that}.draw",
+                excludeSource: "init"
+            }
+        },
+        modelRelay: {
+            source: "{that}.model.dataSet",
+            target: "{that}.model.wrappedDataSet",
+            singleTransform: {
+                type: "fluid.transforms.free",
+                args: ["{that}.model.dataSet"],
+                func: "floe.chartAuthoring.lineChart.chart.wrapSingleDataSet"
+            }
+        },
+        invokers: {
+            draw: {
+                funcName: "floe.chartAuthoring.lineChart.chart.draw",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    // Test if a dataset is multi
+    floe.chartAuthoring.lineChart.chart.isMultiDataSet = function (dataSet) {
+        return dataSet[0].id !== undefined && dataSet[0].data !== undefined;
```

Comment:
This is a pretty unclear treatment of data. This behaviour should be built in by configuration rather than automatic, since it has the potential to be pretty puzzling - that is, there should be explicitly named grades which accept multi datasets and others which don't, and the user should expect to choose which they are dealing with.